services/src/main/java/org/keycloak/theme/DefaultThemeManager.java

```diff
@@ -62,6 +64,14 @@ public Theme getTheme(Theme.Type type) {
         return getTheme(name, type);
     }
 
+    private String typeBasedDefault(Theme.Type type) {
+        if ((type == Theme.Type.ACCOUNT) && isAccount2Enabled) {
+            return "keycloak.v2";
+        }
+        
+        return "keycloak";
+    }
```

Comment:
This code only executes when a theme is set for the realm but it can't find a ThemeProvider.  I'm not even sure how to make that happen, so I'm guessing it would be a bear to write a test.  We have never had a test for that and it's really not that important whether it returns "keycloak.v2" or "keycloak" anyway.  It's just a fallback.